plugins/auth-backend/src/lib/EnvironmentHandler.ts

```diff
@@ -16,49 +16,56 @@
 
 import express from 'express';
 import { AuthProviderRouteHandlers } from '../providers/types';
-import { NotFoundError } from '@backstage/backend-common';
 
 export type EnvironmentHandlers = {
   [key: string]: AuthProviderRouteHandlers;
 };
 
 export class EnvironmentHandler implements AuthProviderRouteHandlers {
-  constructor(private readonly providers: EnvironmentHandlers) {}
+  constructor(
+    private readonly providerId: string,
+    private readonly providers: EnvironmentHandlers,
+  ) {}
 
-  private getProviderForEnv(req: express.Request): AuthProviderRouteHandlers {
+  private getProviderForEnv(
+    req: express.Request,
+    res: express.Response,
+  ): AuthProviderRouteHandlers | undefined {
     const env = req.query.env?.toString();
-    if (!this.providers.hasOwnProperty(env)) {
-      throw new NotFoundError(
-        `No environment for ${env} found in this provider`,
-      );
+
+    if (this.providers.hasOwnProperty(env)) {
+      return this.providers[env];
     }
-    return this.providers[env];
+
+    res.status(404).send(
+      `Missing configuration.
+<br>
+<br>
+For this flow to work you need to supply a valid configuration for the "${env}" environment of the "${this.providerId}" provider.`,
```

Comment:
Assuming that the caller actually renders html. But ok

Reply:
Otherwise they get the newlines and it's clear enough anyway :p